docs/tls/CONFIGURATION.md

```diff
@@ -0,0 +1,183 @@
+```
+
+#### Creating the Client Certificate Secret
+
+This Secret is only required for mutual TLS. It contains the client certificate, and is used by NLK to authenticate to NGINX Plus.
+
+The easiest way to create the secret is by using `kubectl`:
+
+```bash
+kubectl create secret tls nlk-tls-client-secret --cert=/tmp/client.crt --key=/tmp/client.key
+```
+
+#### Secrets as yaml files
+
+The Secrets can also be created using yaml files. The following is an example of a yaml file for the Client Secret:
+
+```yaml
+apiVersion: v1
+data:
+  tls.crt: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUVCVEN...
+  tls.key: LS0tLS1CRUdJTiBQUklWQVRFIEtFWS0tLS0tCk1JSUV2Z0l...
+kind: Secret
+metadata:
+  name: nlk-tls-client-secret
+type: kubernetes.io/tls
+```
+
+Note: While it is possible to generate the data values (`tls.crt` and `tls.key`) manually, the above yaml can be generated using the following command:
+
+```bash
+kubectl create secret tls nlk-tls-client-secret --cert=/tmp/client.crt --key=/tmp/client.key --dry-run=client -o yaml
+```
+
+The same command can be used to generate the yaml for the CA Secret by changing the `--cert` and `--key` filenames.
+
+Once the yaml files are generated they can be applied using `kubectl`:
+
+```bash
+kubectl apply -f ca-secret.yaml
+kubectl apply -f client-secret.yaml
+```
+
+**Warning: it is important that these files do not make their way into a public repository or other storage location where they can be accessed by unauthorized users.**
+
+## Configuring NGINX Plus
+
+### Configuring the Server certificate
+
+Copy the necessary server files (`server.crt`, `server.key`, and `ca.crt`) to the NGINX host; place the files in the `/etc/ssl/certs/nginx` directory.
+
+#### One-way TLS
+
+In the `/etc/nginx/nginx.conf` file, add the following to the `http` or `server` section (refer to the [documentation](https://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_certificate) for details):
+
+```bash
+http {
+  ssl_certificate       /etc/ssl/certs/nginx/server.crt;
+  ssl_certificate_key   /etc/ssl/certs/nginx/server.key;
+}
+```
+
+This will configure NGINX Plus to use the `server.crt` and `server.key` files for TLS.
+
+Reload the NGINX Plus configuration to apply the changes.
+
+```bash
+nginx -s reload
+```
+
+#### Mutual TLS
+
+In the `/etc/nginx/nginx.conf` file, add the following to the `http` or `server` section (refer to the [documentation](https://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_client_certificate) for details):
+
+```bash
+http {
+  ssl_client_certificate    /etc/ssl/certs/nginx/ca.crt;
+  ssl_verify_client         on;
+  ssl_verify_depth          3;
+}
+```
+
+This will configure NGINX Plus to use the `ca.crt` file for client authentication.
+
+Reload the NGINX Plus configuration to apply the changes.
+
+```bash
+nginx -s reload
+```
+
+Test with curl:
+
+```bash
+curl --cert client.crt --key client.key --cacert ca.crt https://<your-host>/api
+```
```

Comment:
Interesting point; other than the fact that no errors are logged and updates are flowing there is no deterministic way I can think of to do this.

Reply:
Yeah I don't have a lot of experience in this area either.  Poking through other tools, it may be a good idea to have some part of NLK other than the configmap validate that it's working in mtls mode when described with kubectl.  Or via some other method.

Another option might be attempting a connection using `openssl` as a client as described in this [post](https://www.golinuxcloud.com/mutual-tls-authentication-mtls/#8_Validate_mutual_TLS_authentication) (method 2)